pennylane/qnode_new/jacobian.py

```diff
@@ -0,0 +1,528 @@
+        var_observables = [e for e in self.circuit.observables if e.return_type == ObservableReturnTypes.Variance]
+
+        # first, replace each var(A) with <A^2>
+        new_observables = []
+        for e in var_observables:
+            # need to calculate d<A^2>/dp
+            w = e.wires
+
+            if self.model == 'qubit':
+                if e.name == 'Hermitian':
+                    # since arbitrary Hermitian observables
+                    # are not guaranteed to be involutory, need to take them into
+                    # account separately to calculate d<A^2>/dp
+
+                    A = e.params[0]  # Hermitian matrix
+                    #if not np.allclose(A @ A, np.identity(A.shape[0])):
+                    new = qml.expval(qml.Hermitian(A @ A, w, do_queue=False))
+                else:
+                    # involutory, A^2 = I
+                    # For involutory observables (A^2 = I) we have d<A^2>/dp = 0
+                    new = qml.expval(qml.Hermitian(np.identity(2**len(w)), w, do_queue=False))
+
+            else:  # CV circuit, first order observable
+                # get the heisenberg representation
+                # This will be a real 1D vector representing the
+                # first order observable in the basis [I, x, p]
+                A = e._heisenberg_rep(e.parameters) # pylint: disable=protected-access
+
+                # take the outer product of the heisenberg representation
+                # with itself, to get a square symmetric matrix representing
+                # the square of the observable
+                A = np.outer(A, A)
+                new = qml.expval(qml.ops.PolyXP(A, w, do_queue=False))
+
+            # replace the var(A) observable with <A^2>
+            self.circuit.update_node(e, new)
+            new_observables.append(new)
+
+        # calculate the analytic derivatives of the <A^2> observables
+        pdA2 = self._pd_parameter_shift(idx, args, kwargs, force_order2=(self.model == 'cv'))  # FIXME the force_order2 use here is convoluted and could be better
+
+        # restore the original observables, but convert their return types to expectation
+        for e, new in zip(var_observables, new_observables):
+            self.circuit.update_node(new, e)
+            e.return_type = ObservableReturnTypes.Expectation
+
+        # evaluate <A>
+        evA = np.asarray(self.evaluate(args, kwargs))
+
+        # evaluate the analytic derivative of <A>
+        pdA = self._pd_parameter_shift(idx, args, kwargs)
+
+        # restore return types
+        for e in var_observables:
+            e.return_type = ObservableReturnTypes.Variance
+
+        # return d(var(A))/dp = d<A^2>/dp -2 * <A> * d<A>/dp for the variances,
+        # d<A>/dp for plain expectations
+        return np.where(where_var, pdA2 -2 * evA * pdA, pdA)
+
+
+
+class AutogradQNode(JacobianQNode):
+    """JacobianQNode that works with Autograd."""
+
+    interface = 'Autograd'
```

Comment:
Should be `interface="numpy"`. I know not technically correct, but this is what we use in the docs, decorator, and new VQE module :)

Reply:
And will also help us slowly deprecate autograd when we need to